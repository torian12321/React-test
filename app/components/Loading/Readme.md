Basic
```
<Loading />
```

Spinner
```
<Loading type='pulse' />
```

Sizes
```
<div>
    <Loading size='xs' />
    <Loading size='sm' />
    <Loading size='md' />
    <Loading size='lg' />
</div>
```

Sizes
```
<div>
    <Button><Loading size='xs' /></Button>
    <Message><Loading /></Message>
</div>
```