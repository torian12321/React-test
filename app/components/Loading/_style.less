@import '~LESS_conf';

.loading{
	@color: @themeColor_1;

    font-size : 32px;
    width     : .5em;
    height    : .5em;
    display   : block;
    margin    : 1em auto;
    background: fadeout(@color, 20%);
    .rounded(50%);
    .animation_pulse(@color, .5em);
    .animation(pulse; 2s);
}

// .loading_bouncing{
//     @color_main: @themeColor_1;
//     @color_sec : @themeColor_2;
//     @gutter    : 1;
    
//     position  : relative;
//     font-size : 10px;
//     text-align: center;
//     margin    : 2em;
//     &,
//     &:after,
//     &:before{
//         height    : 1em;
//         width     : 1em;
//         background: @color_main;
//         box-shadow: 0 0 0 0 @color_sec;
//         .rounded(50%);
//         .animation_bouncing_shadow(@color: @color_sec;);
// 		.animation(bouncing_shadow; 1s; @speedCurve: ease-in-out);
//     }
//     &:after,
//     &:before{
//         content : '';
//         position: absolute;
//         top     : 0;
//     }
//     &:before{
//         left           : (1 + @gutter) * (-1em);
//         animation-delay: -0.2s
//     }
//     &:after{
//         right          : (1 + @gutter) * (-1em);
//         animation-delay: 0.2s
//     }
// }