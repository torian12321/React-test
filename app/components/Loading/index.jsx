var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
styles    = require('./_style.less');

const Loading = props => (
    <div className={classNames('loading', props.type, props.size)} />
);


Loading.propTypes = {
    type: PropTypes.oneOf(['spinner', 'pulse']),
    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg'])
};
Loading.defaultProps = {
    type: 'spinner'
};

module.exports = Loading;
