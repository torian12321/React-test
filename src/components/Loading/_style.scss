@import '../../styles/conf/index.scss';

// Vars
$loading-size         : 32px !default;
$loading-color        : $themeColor_1 !default;
$loading-color-disable: $colorState_disable !default;

// Styles
.loading{
  font-size : $loading-size;
  display   : block;
  opacity   : .8;
  box-sizing: border-box;
  margin    : auto;
  width     : 1em;
  height    : 1em;
  @include rounded-circular();

  &--spinner{
    background      : transparent;
    border          : $loading-color-disable .1em solid;
    border-top-color: $loading-color;
    // .animation_spin();
    // .animation(spin; 1s);
  }
  &--pulse{
    width     : .4em;
    height    : .4em;
    margin    : 1em auto;
    background: $loading-color;
    // .animation_pulse(@color, .5em);
    // .animation(pulse; 1.5s);
  }

  // // Sizes
  &--xs{font-size: $loading-size * 0.5;}
  &--sm{font-size: $loading-size;}
  &--md{font-size: $loading-size * 1.5;}
  &--lg{font-size: $loading-size * 2.5;}
}