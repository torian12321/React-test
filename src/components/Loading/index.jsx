import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class Loading extends React.PureComponent {
  render() {
    return <div className={classNames(
      'loading',
      this.props.type ? `loading--${this.props.type}` : null,
      this.props.size ? `loading--${this.props.size}` : null
    )}/>;
  }
};

Loading.propTypes = {
  type: PropTypes.oneOf(['spinner', 'pulse']),
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg'])
};
Loading.defaultProps = {
  type: 'spinner'
};

module.exports = Loading;
