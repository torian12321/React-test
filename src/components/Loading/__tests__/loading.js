import React    from 'react';
import renderer from 'react-test-renderer';
import Loading  from '../index';

jest.unmock('../index.jsx');

test('Should render Loading', () => {
  const component = renderer.create(
    <Loading />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});