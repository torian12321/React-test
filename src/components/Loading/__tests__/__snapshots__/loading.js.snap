// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Loading 1`] = `
<div
  className="loading spinner"
/>
`;
